src/ppx/metrics.cpp

```diff
@@ -265,20 +342,39 @@ MetricID Manager::AddMetric(const MetricMetadata& metadata)
     if (metric == nullptr) {
         return kInvalidMetricID;
     }
-    auto metricID = mNextMetricID++;
+    metricID = EnsureAllocateID(metricID);
     mActiveMetrics.emplace(metricID, metric);
     return metricID;
 }
 
+MetricID Manager::AddLiveMetric(double halfLife, MetricID metricID)
+{
+    if (metricID != kInvalidMetricID && mLiveMetrics.count(metricID) > 0) {
+        return kInvalidMetricID;
+    }
+    metricID               = EnsureAllocateID(metricID);
+    mLiveMetrics[metricID] = LiveMetric(halfLife);
+    return metricID;
+}
+
 bool Manager::RecordMetricData(MetricID id, const MetricData& data)
 {
+    auto liveMetricIter = mLiveMetrics.find(id);
+    bool hasLiveMetric  = (liveMetricIter != mLiveMetrics.end());
+    if (hasLiveMetric) {
+        liveMetricIter->second.RecordEntry(data);
+    }
     if (mActiveRun == nullptr) {
-        PPX_LOG_WARN("Attempted to record a metric entry with no active run.");
+        if (!hasLiveMetric) {
+            PPX_LOG_WARN("Attempted to record a metric entry with no active run.");
+        }
         return false;
```

Comment:
Seems that the function returns true if the metric data was recorded. If the live metric exits, but there is no active run, shouldn't the function also return true since the live metric got recorded?